### PR TITLE
Support Hunyuan DiT loading

### PR DESCRIPTION
**Pull request by sayakpaul**
I took the liberty to refactor the LoRA related bits in the hooks but LMK if you would like me to discard that. 

Additionally, it might be nice to configure some formatting and linting on the repository. Do you have recommendations? I use this: https://gist.github.com/sayakpaul/a48a0f63d0e346164a97c687f27c5b51.

### CODE REVIEW

File: helpers/sdxl/save_hooks.py [hunk 1]
```diff
@@ -106,79 +113,85 @@ def __init__(
...
+            ):
+                text_encoder_3_lora_layers_to_save = (
+                    convert_state_dict_to_diffusers(
+                        get_peft_model_state_dict(model)
                     )
+                )
 
-                elif isinstance(
+            elif not isinstance(model, type(unwrap_model(self.accelerator, HunyuanDiT2DModel))):
+                if isinstance(
```
Comment (bghira) [inline helpers/sdxl/save_hooks.py:159]: we go from `if isinstance()` to `elif not isinstance()` ? it looks like we never save the hunyuan-dit weights

Comment (sayakpaul) [reply helpers/sdxl/save_hooks.py:159]: Because the codepath below is for LoRA and Hunyuan DiT doesn’t support LoRA yet. Or am I reading it wrong?

**Comment by bghira** [comment]
git pre-commit hook:

```sh
#!/bin/bash

export CHANGED_FILES
CHANGED_FILES=$(git diff --cached --name-only --diff-filter=ACM | grep -e '\.py$')
# Format each changed file:
for file in $CHANGED_FILES; do
    black "$file" || exit 1
    git add "$file"
done

python -m unittest discover -s tests/ || exit 1

exit 0
```

File: helpers/sdxl/save_hooks.py (hunk 1 above)
Comment (bghira) [reply helpers/sdxl/save_hooks.py:159]: since you'd refactored it, i assumed there was support :P

Comment (sayakpaul) [reply helpers/sdxl/save_hooks.py:159]: Hunyuan DiT PEFT support is first needed in diffusers. Once done, will add here.

**Comment by sayakpaul** [comment]
So, it’s just black at the moment. We can move to ruff since it’s faster. Can make a PR for it if you would like?

**Comment by bghira** [comment]
i haven't had any performance issues with black, but i'm not cleaning up 100k files every time either :D

**Comment by sayakpaul** [comment]
Fair. That is understandable.

**Pull request merged**